zerver/tests/test_message_send.py

```diff
@@ -1212,6 +1212,28 @@ def test_send_message_irc_mirror(self) -> None:
         msg = self.get_last_message()
         self.assertEqual(int(datetime_to_timestamp(msg.date_sent)), int(fake_timestamp))
 
+    def test_unsubscribed_send(self) -> None:
+        reset_emails_in_zulip_realm()
+
+        cordelia = self.example_user("cordelia")
+        stream_name = "my_public_stream"
+        self.make_stream(stream_name, invite_only=False)
+        self.unsubscribe(cordelia, stream_name)
+
+        payload = dict(
+            type="stream",
+            to=stream_name,
+            client="test suite",
+            topic="whatever",
+            content="whatever",
+        )
+        cordelia.save()
+        result = self.api_post(cordelia, "/api/v1/messages", payload)
+        self.assert_json_success(result)
+        msg = Message.objects.get(id=result.json()["id"])
+        self.assertEqual(msg.topic_name(), "whatever")
+        self.assertEqual(msg.content, "whatever")
```

Comment:
We probably have an existing backend test for this -- I suppose you could verify by breaking the ability to send while unsubscribed and seeing if a test fails.